standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -9517,6 +9517,9 @@ public Map<String, String> updateTableColumnStatistics(ColumnStatistics colStats
       // So let's not use them anywhere unless absolutely necessary.
       String catName = statsDesc.isSetCatName() ? statsDesc.getCatName() : getDefaultCatalog(conf);
       MTable mTable = ensureGetMTable(catName, statsDesc.getDbName(), statsDesc.getTableName());
+      lockForUpdate("TBLS", "TBL_ID", "\"TBL_ID\" = " + mTable.getId());
```

Comment:
Why do we need a lock on "TBLS" when updating stats?

Reply:
In that scenario, retrying seems like a better approach than placing a global lock on the commonly accessed `TBLS`.

Reply:
As far as I konw it won't prevent others from reading the whole `TBLS`, the cases affected are dropping/altering this `TBL_ID` table.